my-webxr-app/src/components/DataPoint.tsx

```diff
@@ -65,11 +88,11 @@ export default function DataPoint({
       <mesh
         {...meshProps}
         scale={outlineScale || 1.25}
-        visible={hoverCount != 0 || selectedDataPoint === id}
+        visible={hoverCount != 0 || selectedDataPoint?.id === id}
       >
         <sphereGeometry args={size || [0.1, 10, 10]} />
         <meshStandardMaterial
-          color={selectedDataPoint === id ? "blue" : "aqua"}
+          color={selectedDataPoint?.id === id ? "blue" : "aqua"}
```

Comment:
```selectedDataPoint?.id === id```

What is the '?.id' is for? If we want to check for id should 'selectedDataPoint === id' be ok? Is it possible for selectedDataPoint to be null?

Reply:
Yes, in PointSelectionContext the selectedDataPoint can be null if current point is unselected or no points are selected at all.

"?.id" prevents errors if selectedDataPoint is null, and "selectedDataPoint?.id" is for accessing the specific id property from the data point object or else the entire object will be compared to a number type. 

Hope that helps!

Reply:
I see now. I did not see the null in the context file.  Great work!